test/s3select_test.cpp

```diff
@@ -629,3 +693,147 @@ TEST(TestS3selectFunctions, modzero)
+    auto status = s3select_syntax.parse_query(input_query.c_str());
+    ASSERT_EQ(status, 0);
+    s3selectEngine::csv_object s3_csv_object(&s3select_syntax);
+    std::string s3select_result;
+    std::string input;
+    size_t size = 1;
+    generate_csv(input, size);
+    status = s3_csv_object.run_s3select_on_object(s3select_result, input.c_str(), input.size(), 
+        false, // dont skip first line 
+        false, // dont skip last line
+        true   // aggregate call
+        ); 
+    ASSERT_EQ(status, 0); 
+    ASSERT_EQ(s3select_result, std::string("null,\n"));
+} 
+
+TEST(TestS3selectFunctions, nullifstring)
+{
+    s3select s3select_syntax;
+    const std::string input_query = "select nullif(\"james\",\"bond\") from stdin;";
+    auto status = s3select_syntax.parse_query(input_query.c_str());
+    ASSERT_EQ(status, 0);
+    s3selectEngine::csv_object s3_csv_object(&s3select_syntax);
+    std::string s3select_result;
+    std::string input;
+    size_t size = 1;
+    generate_csv(input, size);
+    status = s3_csv_object.run_s3select_on_object(s3select_result, input.c_str(), input.size(), 
+        false, // dont skip first line 
+        false, // dont skip last line
+        true   // aggregate call
+        ); 
+    ASSERT_EQ(status, 0); 
+    ASSERT_EQ(s3select_result, std::string("james,\n"));
+} 
+
+TEST(TestS3selectFunctions, nullifeqstring)
+{
+    s3select s3select_syntax;
+    const std::string input_query = "select nullif(\"redhat\",\"redhat\") from stdin;";
+    auto status = s3select_syntax.parse_query(input_query.c_str());
+    ASSERT_EQ(status, 0);
+    s3selectEngine::csv_object s3_csv_object(&s3select_syntax);
+    std::string s3select_result;
+    std::string input;
+    size_t size = 1;
+    generate_csv(input, size);
+    status = s3_csv_object.run_s3select_on_object(s3select_result, input.c_str(), input.size(), 
+        false, // dont skip first line 
+        false, // dont skip last line
+        true   // aggregate call
+        ); 
+    ASSERT_EQ(status, 0); 
+    ASSERT_EQ(s3select_result, std::string("null,\n"));
+} 
+
+TEST(TestS3selectFunctions, nulladdition)
+{
+    s3select s3select_syntax;
+    const std::string input_query = "select 1 + null from stdin;";
+    auto status = s3select_syntax.parse_query(input_query.c_str());
+    ASSERT_EQ(status, 0);
+    s3selectEngine::csv_object s3_csv_object(&s3select_syntax);
+    std::string s3select_result;
+    std::string input;
+    size_t size = 1;
+    generate_csv(input, size);
+    status = s3_csv_object.run_s3select_on_object(s3select_result, input.c_str(), input.size(), 
+        false, // dont skip first line 
+        false, // dont skip last line
+        true   // aggregate call
+        ); 
+    ASSERT_EQ(status, 0); 
+    ASSERT_EQ(s3select_result, std::string("nan,\n"));
+} 
+
+TEST(TestS3selectFunctions, nullnanarithmetic)
+{
+    s3select s3select_syntax;
+    const std::string input_query = "select 3 * null + nan/5 - 4 * null from stdin;";
```

Comment:
instead of complex query containing several operations.

add test per arithmetic operation (+,/,*-) , with nan/null.
meaning simple binary operation , so we can be sure that all operations are covered.

as in compare operator , write those tests on value object directly.